gemini/use-cases/retrieval-augmented-generation/qna_using_query_routing/utils/qna_vector_search.py

```diff
@@ -0,0 +1,144 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Answer QnA Type Questions using genai Content"""
+
+# Utils
+import configparser
+import logging
+
+from google.cloud import aiplatform
+import pandas as pd
+from vertexai.generative_models import GenerativeModel
+from vertexai.language_models import TextEmbeddingModel
+
+
+class QnAVectorSearch:
+    """genai Generate Answer From genai Content"""
+
+    def __init__(
+        self,
+        model: GenerativeModel,
+        index_endpoint: aiplatform.MatchingEngineIndexEndpoint,
+        deployed_index_id: str,
+        config_file: str,
+        logger=logging.getLogger(),
+    ) -> None:
+        self.config = configparser.ConfigParser()
+        self.config.read(config_file)
+
+        self.logger = logger
+        self.model = model
+        self.index_endpoint = index_endpoint
+        self.deployed_index_id = deployed_index_id
+
+        # Initalizing embedding model
+        self.text_embedding_model = TextEmbeddingModel.from_pretrained(
+            self.config["vector_search"]["embedding_model_name"]
+        )
+
+        self.embedding_df = pd.read_csv(
+            self.config["vector_search"]["embedding_csv_file"]
+        )
```

Comment:
This should really use something like a document store (Firestore can work on Google Cloud) instead of just reading from a CSV